python/delta/integration_tests/unity-catalog-commit-coordinator-integration-tests.py

```diff
@@ -54,54 +63,195 @@
     .config(f"spark.sql.catalog.{CATALOG_NAME}.token", CATALOG_TOKEN) \
     .config(f"spark.sql.catalog.{CATALOG_NAME}.uri", CATALOG_URI) \
     .config(f"spark.sql.defaultCatalog", CATALOG_NAME) \
+    .config("spark.databricks.delta.replaceWhere.constraintCheck.enabled", True) \
     .config("spark.hadoop.fs.s3.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem") \
     .config("spark.databricks.delta.commitcoordinator.unity-catalog.impl",
             "org.delta.catalog.UCCoordinatedCommitClient") \
     .getOrCreate()
 
-expected_error_tag = "UNITY_CATALOG_EXTERNAL_COORDINATED_COMMITS_REQUEST_DENIED"
 
+EXTERNAL_CLIENT_DELETE_ERROR = \
+    f"Request to delete non-external table '{CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE}' " \
+    "from outside of Databricks Unity Catalog enabled compute environment is not supported."
+S3_FORBIDDEN_ACCESS_ERROR = ("Forbidden (Service: Amazon S3; Status Code: 403; "
+                             "Error Code: 403 Forbidden;")
+WRITE_FAILED_ERROR = "[TASK_WRITE_FAILED] Task failed while writing rows to s3"
+
+
+def read(table_name) -> DataFrame:
+    return spark.read.table(f"{CATALOG_NAME}.{SCHEMA}.{table_name}")
+
+
+def read_with_timestamp(timestamp, table_name) -> DataFrame:
+    return spark.read.option("timestampAsOf", timestamp).table(
+        f"{CATALOG_NAME}.{SCHEMA}.{table_name}")
+
+
+def read_with_cdf_timestamp(timestamp, table_name) -> DataFrame:
+    return spark.read.option('readChangeFeed', 'true').option(
+        "startingTimestamp", timestamp).table(f"{CATALOG_NAME}.{SCHEMA}.{table_name}")
+
+
+def create_df_with_rows(list_of_rows) -> DataFrame:
+    return spark.createDataFrame(list_of_rows,
+                                 schema=StructType([StructField("id", IntegerType(), True)]))
+
+
+class UnityCatalogCommitCoordinatorTestSuite(unittest.TestCase):
+    setup_df = spark.createDataFrame([(1, ), (2, ), (3, )],
+                                     schema=StructType([StructField("id", IntegerType(), True)]))
+
+    def setUp(self) -> None:
+        self.setup_df.write.mode("overwrite").insertInto(
+            f"{CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE}")
```

Comment:
You could define MANAGED_CC_TABLE = f"{CATALOG_NAME}.{SCHEMA}.{MANAGED_CC_TABLE_NAME}" to reduce verbosity